### PR TITLE
#167727620 Integrate Coveralls Code Coverage Service

### PR DESCRIPTION
**Pull request by ChidozieGabriel**
#### What does this PR do?
- Integrate coveralls code coverage service.

#### Description of Task to be completed?
- Add this repository to [coveralls.io](https://coveralls.io/github/andela/rogue-backend)
- Add coveralls [badge](https://img.shields.io/coveralls/github/andela/rogue-backend/develop.svg?style=plastic) to README.md

#### What are the relevant pivotal tracker stories?

[#167727620](https://www.pivotaltracker.com/story/show/167727620)

### CODE REVIEW

**Review by 0xciph3r — CHANGES_REQUESTED** [review]
The develop branch has been updated. Kindly ensure those changes are reflecting on this PR

**Comment by ChidozieGabriel** [comment]
I have resolved the conflicts.

**Review by 0xciph3r — APPROVED** [review]

**Comment by 0xciph3r** [comment]
Sorry, you have a mistake in your package.json file. A missing comma

**Review by 0xciph3r — CHANGES_REQUESTED** [review]
There is a missing comma in your package.json file

**Review by Daymorelah — CHANGES_REQUESTED** [review]
Has this feature been tested with actual test coverage sent to coveralls?

**Comment by ChidozieGabriel** [comment]
@Daymorelah No. It has not been tested. This is because there are no tests yet in the codebase.
Please do I have to write tests for it?

**Comment by ChidozieGabriel** [comment]
@NonsoAmadi10 I have fixed the missing comma. Thanks.

**Review by 0xciph3r — APPROVED** [review]

**Comment by Daymorelah** [comment]
> @Daymorelah No. It has not been tested. This is because there are no tests yet in the codebase.
> Please do I have to write tests for it?

There is a chore for setting up unit testing for the app. Be on the lookout for when it is merged so that your chore can be tested.

**Comment by ChidozieGabriel** [comment]
@Daymorelah Okay. Note taken.
Thank you.

**Review by 0xciph3r — APPROVED** [review]
Update the changes from the develop branch

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
+    "coverage": "nyc report --reporter=text-lcov | coveralls",
```
Comment (Daymorelah) [inline package.json:7]: Has coveralls been able to display coverage data?

Comment (ChidozieGabriel) [reply package.json:7]: Yes.
https://coveralls.io/github/andela/rogue-backend?branch=ch-integrate-coveralls-167727620

File: src/utils/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const add = (a, b) => a + b;
```
[the remark this answers is not in the document]
Comment (ChidozieGabriel) [reply src/utils/index.js:1]: ` 'sourceType: module'` will be added with the ESLint configuration.
Thank you Mr hound.

**Comment by ChidozieGabriel** [comment]
@Daymorelah The coveralls service has been tested; it works.
https://coveralls.io/github/andela/rogue-backend?branch=ch-integrate-coveralls-167727620
Thanks.

File: .coveralls.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+repo_token: YgA8hpToZwO7tMdm7owflbzfogJ7Z4mzQ
```
Comment (Daymorelah) [inline .coveralls.yml]: Why are you exposing the repo token?

Comment (Daymorelah) [reply .coveralls.yml]: This should be hidden by using an environment variable. use this environment variable in Travis `CODECLIMATE_REPO_TOKEN`. and set the token as its value there

**Review by Daymorelah — CHANGES_REQUESTED** [review]

File: .coveralls.yml (hunk 3 above)
Comment (ChidozieGabriel) [reply .coveralls.yml]: Okay. I've resolved it.
Thanks.

**Review by Daymorelah — APPROVED** [review]

**Pull request merged**